src/main/java/org/opensearch/security/transport/SecurityInterceptor.java

```diff
@@ -127,6 +128,8 @@ public <T extends TransportResponse> void sendRequestDecorate(AsyncSender sender
         final String origCCSTransientMf = getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_MASKED_FIELD_CCS);
 
         final boolean isDebugEnabled = log.isDebugEnabled();
+        final boolean isDirectRequest = HeaderHelper.isDirectRequest(getThreadContext());
+
         try (ThreadContext.StoredContext stashedContext = getThreadContext().stashContext()) {
```

Comment:
Could not yet find a solution to not stash the context for same node.

Tried with these changes:
```
try (ThreadContext.StoredContext stashedContext = (isDirectRequest ? null: getThreadContext().stashContext())) {
    final TransportResponseHandler<T> restoringHandler = isDirectRequest ? handler : new RestoringTransportResponseHandler<T>(handler, stashedContext);
```

But the test fails with ClusterManagernotDiscoveredException everytime on test start.



```

Reply:
It is perhaps good idea to continue stashing from security point of view too. That way, any existing transients assumed to be cleared are not carry forwarded.

For example let us assume current thread context had a,b,c entries and we explicitly passed around only a & b. Now carry forwarding c may be a security concern.

Reply:
> But the test fails with ClusterManagernotDiscoveredException everytime on test start.

I would guess that this is related to the wrong check for `isDirectRequest`. But, yes, I would also recommend to stash anyway and just restore the transients which you want to keep. That would be the most conservative and most minimal change.

Reply:
should we say `isLocalNodeRequest` to add more readability?  or are we categorizing local node request to  `direct` channel through out the code-base ?